lib/ohai/plugins/solaris2/cpu.rb

```diff
@@ -38,4 +34,35 @@
       cpu[index]['mhz'] = cpu_info[10]
     end
   end
+
+  def get_sparc_processor_info
+    i = 0
+    cores = 0
+    shell_out('psrinfo -v -p').stdout.lines.each do |line|
+      case  line.strip
+        when /(\d+)\s+cores/
+          cores += "#{$1}".to_i
+        when /^(\S+).*\b(\d+)\s+MHz\)$/
+          index = i.to_s
+          cpu[index] = Mash.new
+          cpu[index]['model_name'] = "#{$1}"
```

Comment:
here too; just `[model_name] = $1`. Also, please consistently use either single quotes or double quotes. My preference would be to use doubles.